src/Tasks/CodeTaskFactory.cs

```diff
@@ -851,20 +847,44 @@ private Assembly CompileInMemoryAssembly()
         /// <summary>
         /// Combine our default referenced assemblies with those explicitly specified
         /// </summary>
-        private void CombineReferencedAssemblies(List<string> finalReferenceList)
+        private List<string> CombineReferencedAssemblies()
```

Comment:
Do CodeTaskFactorys get reused? If so, this removed the caching, so there could be a perf hit.

Reply:
Since the previous cache `s_defaultReferencedAssemblies` applied to the raw inputs and it mapped those to filenames again every time (via probing/reflection) I don't think the lack of the input cache is likely to be significant.